server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSWebOptionsCheckUtility.java

```diff
@@ -397,6 +412,12 @@ private void checkRead(CICSParser.Cics_web_readContext ctx) {
 
     checkPrerequisiteIsMet(
         ctx.FORMFIELD(), ctx.CHARACTERSET(), ctx, "CHARACTERSET without FORMFIELD");
+    if (noLengthOptionsEnabled()
+        && (!ctx.FORMFIELD().isEmpty()
+            || !ctx.HTTPHEADER().isEmpty()
+            || !ctx.QUERYPARM().isEmpty())) {
+      checkHasMandatoryOptions(ctx.NAMELENGTH(), ctx, "NAMELENGTH");
+    }
```

Comment:
at least one of those operands is required.
```suggestion
    if (noLengthOptionsEnabled()) {
      checkHasMandatoryOptions(ctx.NAMELENGTH(), ctx, "NAMELENGTH");
    }
```